### ISSUE TITLE
SDL2: fingerId Android vs iOS differences

### ISSUE THREAD

**Issue by DominusExult** [libsdl-org/SDL#11667]
Current SDL2 git, Android SDK 34 (minTarget 28), iOS 18.2

On Android logging `event.tfinger.fingerId` returns the numbers of fingers touching the screen, beginning with `0`.

On iOS, `fingerId` returns some arbitrary(?) number, e.g. 4553986240. And when a connected mouse is used it returns `0` on clicking.

On Exult we were doing something on `fingerId != 0`, as I assumed that 0 = is the mouse touch. I am mostly on macOS and iOS and only lately were looking more into our Android port and was surprised that this code is not working correctly.

Is this on purpose or a bug?

**Comment by slime73** [comment]
You're supposed to use `SDL_MOUSE_TOUCHID` for recognizing a mouse touch device: https://github.com/libsdl-org/SDL/blob/SDL2/include/SDL_touch.h#L64

It's a touch device ID rather than a finger ID.

Touch finger IDs are arbitrary IDs whose actual value depends on the backend and doesn't mean anything important for API users.

**Comment by DominusExult** [comment]
`SDL_MOUSE_TOUCHID` returns to me the same value for finger touch and mouse click (`-1`).
However if I use `event.tfinger.touchId` I get a reliable result (mouse 0, finger 1 on both iOS and Android).

The use case is that on our ports for iOS and Android we hide the mouse pointer. Unless you are using a connected mouse, then we detect mouse motion via `SDL_TOUCH_MOUSEID`.
The tricky part is to recognize when we switch from mouse to touch again. 
Coming from my iOS results, I assumed `fingerId` for mouse is always 0 and real finger touches an arbitrary number (which did make sense). That worked on iOS and I never tried anything else. Bad assumption on my part.

Question: not that I have any use for it but is it reliable that the not-so-arbitrary arbitrary number on Android returns the number of fingers touching?
And wouldn't it make sense for iOS to use this as well (or change Android accordingly that it really is an arbitrary number)?

**Comment by slime73** [comment]
Maybe it'd be less confusing if a finger id was a completely opaque type.

You can use `SDL_GetNumTouchFingers` if you want to count the number of active touch presses for a device, inspecting the finger ids won't give you anything you can rely on, it just happens that Android's current arbitrary implementation looks similar to that info.

**Comment by Ghabry** [comment]
This is not just a Android vs iOS difference.

I also noticed the same on Windows with two different devices:

On my Surface the Finger ID goes from 1-5 depending on how many fingers are on the screen (so like on Android)

But on my new ASUS device the Finger ID _increments_ on every touch. So after 100 touches you get finger ID 100.

-----

Is this documented anywhere that the IDs are arbitrary? Because I felt into the same trap when I implemented this the first time only testing on Android and my Surface :sweat_smile:
